Lab1 Beta 1

DaneWoods opened the pull request:
Info is in the docs folder file.

DaneWoods committed:
Update README.md

DaneWoods committed:
Update README.md

DaneWoods committed:
Update README.md

DaneWoods committed:
add the new concepts so that i can use other validation methods

DaneWoods committed:
Merge branch 'Lab1' of https://github.com/DaneWoods/CS-296N-Labs into Lab1

DaneWoods committed:
make the comment page take a model and the controller accept the com model

DaneWoods committed:
add 2 new properties for validations

DaneWoods committed:
Update README.md

DaneWoods committed:
Update README.md

DaneWoods committed:
Update README.md

DaneWoods committed:
Delete README.md

DaneWoods committed:
Added info file

DaneWoods committed:
Merge branch 'Lab1' of https://github.com/DaneWoods/CS-296N-Labs into Lab1

JacobFelten commented:
# Code Review Form

|                                                      | ---------------------------------------             |
| ---------------------------------------------------- | --------------------------------------------------- |
| Course  number, lab number and lab group             | CS 296N, Lab 1                                      |
| Developer                                            | Dane Woods                                          |
| URL  for the project repository and branch on GitHub | https://github.com/DaneWoods/CS-296N-Labs/tree/Lab1 |
| URL  on a server (if it has been published)          |                                                     |
| Reviewer  and Date                                   | Jacob Felten 1/14/2020                              |

###  Instructions

The reviewer will complete this form for the beta version of a lab assignment done by one of their lab partners. After filling out the “Beta” column and adding comments, the reviewer will submit this document to the Code Review assignment on the LMS.

The developer will revise the beta version of their lab work and fill out the “Production” column to reflect any changes they have made. The developer will submit this completed form along with the production version of their lab assignment.

### Review

| **Criteria**                                                 | **Beta** | **Release** |
| ------------------------------------------------------------ | -------- | :---------- |
| Does  it compile and run without errors?                     | Yes      |             |
|                                                              |          |             |
| Do  all the pages load correctly?                            | Yes      |             |
|                                                              |          |             |
| Does  the style conform to MVC conventions and our class standards? | Yes      |             |
|                                                              |          |             |
| Do  all the links, buttons or other UI elements work correctly? | Yes      |             |
|                                                              |          |             |
|                                                              |          |             |
| Do  the design and implementation conform to OOP best practices? | Yes      |             |
|                                                              |          |             |
|                                                              |          |             |
| Does  the style conform to C# coding conventions?            | Yes      |             |
|                                                              |          |             |
|                                                              |          |             |
| Does  the solution meet all the requirements? (list any issues below) | No       |             |
| Currently there are only 4 properties with annotations instead of 6. |          |             |
|                                                              |          |             |
|                                                              |          |             |
|                                                              |          |             |
|                                                              |          |             |

#### Comments:

 Rating stories with comments works great!





------

 

## Appendix

### Aspects of coding style to check

- Is proper indentation used?
- Are the HTML elements and variables named descriptively?
- Have any unnecessary lines of code or files been removed?
- Are there explanatory comments in the code?
- Do variable names use camelCase? 
- Are properties, methods and classes named using PascalCase (aka TitleCase)?
- Are constant names written using ALL_CAPS?

### Best practices in Object Oriented Programming

- Is the code DRY (no duplicated blocks of code)?
- Are named constants used instead of repeated literal constants?
- Is code that does computation or logical operations separated into its own class instead of being added to the code-behind?
- Are all instance variables private?
- Are local variables used instead of instance variables wherever possible?
- Does each method do just one thing (no “Swiss Armey” methods)?
- Are classes “loosely coupled” and “highly coherent”?

 

------

Written by Brian Bird, Lane Community College, winter 2017, updated winter 2020

------

DaneWoods committed:
added a starter user model, and every little thing needed to functionally implament it

DaneWoods commented:
  | ---------------------------------------
-- | --
Course number, lab number and lab group | CS 296N, Lab 1
Developer | Dane Woods
URL for the project repository and branch on GitHub | https://github.com/DaneWoods/CS-296N-Labs/tree/Lab1
URL on a server (if it has been published) |  
Reviewer and Date | Jacob Felten 1/14/2020

DaneWoods commented:
### Code Review Form
  | ---------------------------------------
-- | --
Course number, lab number and lab group | CS 296N, Lab 1
Developer | Dane Woods
URL for the project repository and branch on GitHub | https://github.com/DaneWoods/CS-296N-Labs/tree/Lab1
URL on a server (if it has been published) |  
Reviewer and Date | Jacob Felten 1/14/2020


### Instructions

The reviewer will complete this form for the beta version of a lab assignment done by one of their lab partners. After filling out the “Beta” column and adding comments, the reviewer will submit this document to the Code Review assignment on the LMS.

The developer will revise the beta version of their lab work and fill out the “Production” column to reflect any changes they have made. The developer will submit this completed form along with the production version of their lab assignment.
### Review
Criteria | Beta | Release
-- | -- | --
Does it compile and run without errors? | Yes | yes
  |   |  
Do all the pages load correctly? | Yes | yes
  |   |  
Does the style conform to MVC conventions and our class standards? | Yes | yes
  |   |  
Do all the links, buttons or other UI elements work correctly? | Yes | yes
  |   |  
  |   |  
Do the design and implementation conform to OOP best practices? | Yes | yes
  |   |  
  |   |  
Does the style conform to C# coding conventions? | Yes | yes
  |   |  
  |   |  
Does the solution meet all the requirements? (list any issues below) | No | yes
Currently there are only 4 properties with annotations instead of 6. |   |  
Added a user model with 2 new properties and added annotation, it is the beginning of the next chapter for authentication, and authorizing. The user model has unit test for the controller, has a table, method for adding a user, etc. |   |  
  |   |  
  |   |  
  |   |  
#### Comments:

Rating stories with comments works great!

------

## Appendix

### Aspects of coding style to check

- Is proper indentation used?
- Are the HTML elements and variables named descriptively?
- Have any unnecessary lines of code or files been removed?
- Are there explanatory comments in the code?
- Do variable names use camelCase?
- Are properties, methods and classes named using PascalCase (aka TitleCase)?
- Are constant names written using ALL_CAPS?
### Best practices in Object Oriented Programming

- Is the code DRY (no duplicated blocks of code)?
- Are named constants used instead of repeated literal constants?
- Is code that does computation or logical operations separated into its own class instead of being added to the code-behind?
- Are all instance variables private?
- Are local variables used instead of instance variables wherever possible?
- Does each method do just one thing (no “Swiss Armey” methods)?
- Are classes “loosely coupled” and “highly coherent”?

------

Written by Brian Bird, Lane Community College, winter 2017, updated winter 2020

DaneWoods committed:
Updated the typora file in the docs folder
